src/components/keyboard/Keyboard.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { Dimensions, TouchableOpacity, StyleSheet, Text, View, ViewStyle } from 'react-native';
+import { useTheme } from '../../styles/themes';
+import { Icon } from '../icons/Icon';
+import type { IconName } from '../icons/IconProps';
+const { height: SCREEN_HEIGHT } = Dimensions.get('screen');
+
+type KeyboardAction = 'type' | 'delete' | 'submit';
+interface KeyboardData {
+    value?: string;
+    action: KeyboardAction;
+    icon?: IconName;
+    iconColor?: string;
+}
+
+interface Props {
+    inputValue: string;
+    setInputValue: (text: string) => void;
+    onSubmit: () => void;
+    disabled?: boolean;
+    style?: ViewStyle;
+}
+
+export const Keyboard = ({ inputValue, setInputValue, onSubmit, disabled, style }: Props) => {
+    const theme = useTheme();
+
+    const keyboardData: KeyboardData[] = [
```

Comment:
The layout of the Keyboard is not expected to change or be dynamic in any way, so I think the implementation should reflect that and go with a classic declarative approach do define the layout.

It will be easier to reason about in many ways; something like

```html
<View style={styles.container}>
    <KeyboardRow>
          <NumberButton value="1" onPress={onInput('1')} />
          <NumberButton value="2"  onPress={onInput('2')} />
          <NumberButton value="3"  onPress={onInput('3')} /> 
    </KeyboardRow>

    <KeyboardRow>
          <NumberButton value="4"   onPress={onInput('4')} />
          <NumberButton value="5"   onPress={onInput('5')} />
          <NumberButton value="6"  onPress={onInput('6')} /> 
    </KeyboardRow>

     <KeyboardRow>
           <NumberButton value="7"  onPress={onInput('7')} />
          <NumberButton value="8"  onPress={onInput('8')} />
          <NumberButton value="9"  onPress={onInput('9')} /> 
    </KeyboardRow>

    <KeyboardRow>
        <DeleteButton onPress={onDelete}/>
        <NumberButton value="0"   onPress={onInput('0')} />
        <SubmitButton onPress={onSubmit}/>
    </KeyboardRow>
</View>
```

Each internal component (`KeyboardRow`, `NumberButton`, `DeleteButton`, `SubmitButton`) is responsable for his on styling.


What do you think about it ?

Reply:
okay, i will apply it , thanks

Reply:
Just my 2 cents, I like the ability to provide a custom matrix to customise the keyboard like in the following project https://github.com/lukebrandonfarrell/react-native-screen-keyboard